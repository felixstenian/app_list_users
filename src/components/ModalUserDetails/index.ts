export { default } from './ModalUserDetails'
export type { ModalUserDetailsProps } from './ModalUserDetails'
