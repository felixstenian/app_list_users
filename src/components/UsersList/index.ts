export { default as CardUser } from './CardUser'
export type { CardUserProps } from './CardUser'
export { default } from './UsersList'
export type { UsersListProps } from './UsersList'
