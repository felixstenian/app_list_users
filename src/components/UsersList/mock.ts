export const listMock = [
  {
    gender: 'female',
    name: {
      title: 'Miss',
      first: 'Jennie',
      last: 'Nichols'
    },
    location: {
      street: {
        number: 8929,
        name: 'Valwood Pkwy'
      },
      city: 'Billings',
      state: 'Michigan',
      country: 'United States',
      postcode: '63104',
      coordinates: {
        latitude: '-69.8246',
        longitude: '134.8719'
      },
      timezone: {
        offset: '+9:30',
        description: 'Adelaide, Darwin'
      }
    },
    email: 'jennie.nichols@example.com',
    login: {
      uuid: '7a0eed16-9430-4d68-901f-c0d4c1c3bf00',
      username: 'yellowpeacock117',
      password: 'addison',
      salt: 'sld1yGtd',
      md5: 'ab54ac4c0be9480ae8fa5e9e2a5196a3',
      sha1: 'edcf2ce613cbdea349133c52dc2f3b83168dc51b',
      sha256: '48df5229235ada28389b91e60a935e4f9b73eb4bdb855ef9258a1751f10bdc5d'
    },
    dob: {
      date: '1992-03-08T15:13:16.688Z',
      age: 30
    },
    registered: {
      date: '2007-07-09T05:51:59.390Z',
      age: 14
    },
    phone: '(272) 790-0888',
    cell: '(489) 330-2385',
    id: {
      name: 'SSN',
      value: '405-88-3636'
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/75.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/75.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/75.jpg'
    },
    nat: 'US'
  },
  {
    gender: 'female',
    name: {
      title: 'Mrs',
      first: 'Ruby',
      last: 'Jones'
    },
    location: {
      street: {
        number: 4803,
        name: 'Wiri Station Road'
      },
      city: 'Greymouth',
      state: 'Wellington',
      country: 'New Zealand',
      postcode: 17397,
      coordinates: {
        latitude: '54.2932',
        longitude: '157.3572'
      },
      timezone: {
        offset: '+5:30',
        description: 'Bombay, Calcutta, Madras, New Delhi'
      }
    },
    email: 'ruby.jones@example.com',
    login: {
      uuid: 'd55b7f3e-1330-41e5-9582-acd000035cf1',
      username: 'blackmouse796',
      password: 'kristina',
      salt: 'nmcf5uwv',
      md5: '2553b4d72b2ea7eee593b5791a6054bf',
      sha1: 'dc022ea1291986b281d7f21395ebffecf343db77',
      sha256: '7bc81c3c2198f5eb2c3030a6bf8d62cc8b4c432357e5e0f7c1b5f2395e5987a0'
    },
    dob: {
      date: '1993-08-10T17:00:07.860Z',
      age: 29
    },
    registered: {
      date: '2003-01-23T06:05:41.680Z',
      age: 20
    },
    phone: '(002)-379-0454',
    cell: '(311)-839-3611',
    id: {
      name: '',
      value: null
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/women/2.jpg',
      medium: 'https://randomuser.me/api/portraits/med/women/2.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/women/2.jpg'
    },
    nat: 'NZ'
  }
]
