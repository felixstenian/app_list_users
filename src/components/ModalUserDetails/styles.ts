import styled from 'styled-components'

import { Row } from 'components'

export const Divider = styled(Row)`
  justify-content: flex-start;
  width: 100%;
  border-bottom: 1px solid;
  border-color: ${({ theme }) => theme.colors.gray};
  padding: 1rem;
`
export const Column = styled(Row)`
  flex-direction: column;
  align-items: flex-start;
`
