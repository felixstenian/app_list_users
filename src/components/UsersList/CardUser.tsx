import { useState } from 'react'

import { Image, ModalUserDetails, Row, Text } from 'components'
import { User } from 'shared/types'

export type CardUserProps = {
  userData: User
}

const CardUser = ({ userData }: CardUserProps) => {
  const [openProfile, setOpenProfile] = useState<boolean>(false)

  const displayDataUser = () => setOpenProfile(true)
  const onCloseUserDetails = () => setOpenProfile(false)

  return (
    <>
      {openProfile && (
        <ModalUserDetails
          userData={userData}
          open={openProfile}
          onClose={onCloseUserDetails}
        />
      )}
      <Row
        key={userData?.login?.uuid}
        bg="secondary"
        width="80%"
        justifyContent="center"
        alignItems="center"
        p={10}
        borderRadius="8px"
        my="5px"
        gap="1rem"
        flexWrap="wrap"
        minWidth="32rem"
        as="a"
        onClick={displayDataUser}
        cursor="pointer"
      >
        <Row
          alignItems="center"
          justifyContent={['center', 'flex-start', 'flex-start']}
          flex={1}
        >
          <Image
            src={userData?.picture?.large}
            alt="alt"
            width={['10rem', '12rem', '8rem']}
            height={['10rem', '12rem', '8rem']}
            borderRadius="10rem"
          />
          <Text ml={[30, 20, 20]} variant="mediumTitle">
            {userData?.name?.first}
          </Text>
        </Row>
        <Row
          flexWrap="wrap"
          justifyContent="center"
          gap={['4rem', '3rem', '5rem']}
        >
          <Text variant="subtitle">{userData?.location?.country}</Text>
          <Text>{userData?.email}</Text>
        </Row>
      </Row>
    </>
  )
}

export default CardUser
