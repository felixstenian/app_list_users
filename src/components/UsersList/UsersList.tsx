import { Row } from 'components'

import { CardUser, User } from '.'

export type UsersListProps = {
  list: User[]
}

const UsersList = ({ list }: UsersListProps) => (
  <Row flexDirection="column" px={10}>
    {list.map((user) => (
      <CardUser key={user.id} user={user} />
    ))}
  </Row>
)

export default UsersList
