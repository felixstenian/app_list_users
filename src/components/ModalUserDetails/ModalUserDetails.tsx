import Link from 'next/link'
import { GrMapLocation } from 'react-icons/gr'

import { Image, Modal, Row, Text } from 'components'
import { formatIsoStringToDateString } from 'shared/helpers'
import { User } from 'shared/types'

import * as Styled from './styles'

export type ModalUserDetailsProps = {
  userData: User
  open: boolean
  onClose: () => void
}

const ModalUserDetails = ({
  userData,
  open,
  onClose
}: ModalUserDetailsProps) => {
  return (
    <Modal
      open={open}
      hasCloseIcon
      onClose={onClose}
      width={['40rem', '40rem', '80rem']}
      height={['65rem', '67rem', '60rem']}
      m={[2, 2, 30]}
      overflowY="auto"
    >
      <Row alignItems="center" flexDirection="column" gap="1rem">
        <Row flexDirection="column" flexWrap="wrap" alignItems="center">
          <Image
            src={userData?.picture?.large}
            alt="alt"
            width="15rem"
            borderRadius="10rem"
          />
          <Row mt={2}>
            <Text variant="body" mr={2}>
              {userData?.name?.title}
            </Text>
            <Text variant="biggerTitle">{`${userData?.name?.first} ${userData?.name?.last}`}</Text>
          </Row>
        </Row>

        <Styled.Divider>
          <Text variant="mediumTitle">Dados</Text>
        </Styled.Divider>

        <Row gap={['2rem', '2rem', '8rem']} flexWrap="wrap">
          <Styled.Column>
            <Text variant="body">Email:</Text>
            <Text variant="mediumTitle">{userData?.email}</Text>
          </Styled.Column>

          <Styled.Column>
            <Text variant="body">Data de Nascimento:</Text>
            <Text variant="mediumTitle">
              {formatIsoStringToDateString(userData?.dob?.date)}
            </Text>
          </Styled.Column>
          <Styled.Column>
            <Text variant="body">Idade:</Text>
            <Text variant="mediumTitle">{userData?.dob?.age}</Text>
          </Styled.Column>
        </Row>

        <Row gap={['2rem', '2rem', '8rem']} flexWrap="wrap">
          <Styled.Column>
            <Text variant="body">Telefone:</Text>
            <Text variant="mediumTitle">{userData?.phone}</Text>
          </Styled.Column>
          <Styled.Column>
            <Text variant="body">Celular:</Text>
            <Text variant="mediumTitle">{userData?.cell}</Text>
          </Styled.Column>
          <Styled.Column>
            <Text variant="body">Data de registro:</Text>
            <Text variant="mediumTitle">
              {formatIsoStringToDateString(userData?.registered?.date)}
            </Text>
          </Styled.Column>
        </Row>

        <Styled.Divider>
          <Text variant="mediumTitle">Endereço</Text>
        </Styled.Divider>

        <Row gap={['2rem', '2rem', '8rem']} flexWrap="wrap">
          <Styled.Column>
            <Text variant="body">País:</Text>
            <Text variant="mediumTitle">{userData?.location?.country}</Text>
          </Styled.Column>
          <Styled.Column>
            <Text variant="body">Estado:</Text>
            <Text variant="mediumTitle">{userData?.location?.state}</Text>
          </Styled.Column>
          <Styled.Column>
            <Text variant="body">Cidade:</Text>
            <Text variant="mediumTitle">{userData?.location?.city}</Text>
          </Styled.Column>
        </Row>

        <Row gap={['2rem', '2rem', '8rem']} flexWrap="wrap">
          <Styled.Column>
            <Text variant="body">Rua:</Text>
            <Text variant="mediumTitle">
              {`${userData?.location?.street?.name}, ${userData?.location?.street?.number}`}
            </Text>
          </Styled.Column>
          <Row
            alignItems="flex-start"
            flexDirection="column"
            justifyContent="center"
          >
            <Link
              passHref
              href={`https://www.google.com/maps/search/?api=1&query=${parseFloat(
                userData?.location?.coordinates?.latitude
              )},${parseFloat(userData?.location?.coordinates?.longitude)}
`}
            >
              <a target="_blank" rel="noopener noreferrer">
                <GrMapLocation size={28} color="#fafa" />
              </a>
            </Link>
          </Row>
        </Row>
      </Row>
    </Modal>
  )
}

export default ModalUserDetails
