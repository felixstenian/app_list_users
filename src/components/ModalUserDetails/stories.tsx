import { Meta, Story } from '@storybook/react'

import { listMock } from 'components/UsersList/mock'
import ModalUserDetails, { ModalUserDetailsProps } from '.'

export default {
  title: 'ModalUserDetails',
  component: ModalUserDetails,
  args: {
    userData: {
      type: listMock[1]
    },
    open: {
      type: true
    }
  }
} as Meta

export const Default: Story<ModalUserDetailsProps> = (args) => (
  <ModalUserDetails {...args} />
)

Default.args = {
  userData: listMock[1]
}
