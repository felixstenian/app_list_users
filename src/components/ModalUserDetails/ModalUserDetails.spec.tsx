import { fireEvent, screen } from '@testing-library/react'

import { formatIsoStringToDateString } from 'shared/helpers'
import { renderWithTheme } from 'shared/utils'
import { listMock } from 'components/UsersList/mock'
import ModalUserDetails from '.'

describe('<ModalUserDetails />', () => {
  it('1. should render the modal with the correct data', () => {
    const { container } = renderWithTheme(
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      <ModalUserDetails userData={listMock[0]} open={true} onClose={() => {}} />
    )

    expect(screen.getByText(listMock[0].name.title)).toBeInTheDocument()
    expect(
      screen.getByText(`${listMock[0].name.first} ${listMock[0].name.last}`)
    ).toBeInTheDocument()
    expect(screen.getByText(listMock[0].email)).toBeInTheDocument()
    expect(
      screen.getByText(formatIsoStringToDateString(listMock[0].dob.date))
    ).toBeInTheDocument()
    expect(screen.getByText(listMock[0].dob.age)).toBeInTheDocument()
    expect(screen.getByText(listMock[0].location.country)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('2. should call the onClose function when the close icon is clicked', () => {
    const onCloseMock = jest.fn()
    renderWithTheme(
      <ModalUserDetails
        userData={listMock[0]}
        open={true}
        onClose={onCloseMock}
      />
    )

    fireEvent.click(screen.getByTestId('modal-close-icon'))
    expect(onCloseMock).toHaveBeenCalledTimes(1)
  })
})
