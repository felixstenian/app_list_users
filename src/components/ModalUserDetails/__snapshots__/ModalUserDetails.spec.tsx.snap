// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalUserDetails /> 1. should render the modal with the correct data 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 8px;
  width: 40rem;
  height: 65rem;
  overflow-y: auto;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 8px;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
  margin-right: 8px;
}

.c11 {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 2.8rem;
}

.c13 {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 2.0rem;
}

.c16 {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c7 {
  width: 15rem;
  width: 15rem;
  border-radius: 10rem;
  overflow: hidden;
}

.c8 {
  width: 100%;
  height: 100%;
}

.c1 {
  width: 100%;
  height: 100%;
  top: 0;
  bottom: 0;
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  z-index: 40;
  background: rgba(0,0,0,0.25);
}

.c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 400px;
  margin-left: 24px;
  margin-right: 24px;
  padding: 24px;
  border-radius: 8px;
  background-color: white;
}

.c4 {
  position: absolute;
  top: 24px;
  right: 24px;
  background: none;
  cursor: pointer;
}

.c12 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  border-bottom: 1px solid;
  border-color: #8F8F8F;
  padding: 1rem;
}

.c15 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

@media screen and (min-width:40em) {
  .c2 {
    margin: 8px;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    margin: 30px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    width: 40rem;
    height: 67rem;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    width: 80rem;
    height: 60rem;
  }
}

@media screen and (min-width:40em) {
  .c14 {
    gap: 2rem;
  }
}

@media screen and (min-width:52em) {
  .c14 {
    gap: 8rem;
  }
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3"
    height="65rem,67rem,60rem"
    width="40rem,40rem,80rem"
  >
    <div
      class="c0 c4"
      data-testid="modal-close-icon"
    >
      <svg
        color="black"
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        style="color: black;"
        viewBox="0 0 1024 1024"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 0 0-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
        />
        <path
          d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
      </svg>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <figure
          class="c7"
          width="15rem"
        >
          <img
            alt="alt"
            class="c8"
            src="https://randomuser.me/api/portraits/men/75.jpg"
          />
        </figure>
        <div
          class="c9"
        >
          <paragraph
            class="c10"
          >
            Miss
          </paragraph>
          <h1
            class="c11"
          >
            Jennie Nichols
          </h1>
        </div>
      </div>
      <div
        class="c0 c12"
      >
        <h2
          class="c13"
        >
          Dados
        </h2>
      </div>
      <div
        class="c14"
      >
        <div
          class="c0 c15"
        >
          <paragraph
            class="c16"
          >
            Email:
          </paragraph>
          <h2
            class="c13"
          >
            jennie.nichols@example.com
          </h2>
        </div>
        <div
          class="c0 c15"
        >
          <paragraph
            class="c16"
          >
            Data de Nascimento:
          </paragraph>
          <h2
            class="c13"
          >
            08/03/1992
          </h2>
        </div>
        <div
          class="c0 c15"
        >
          <paragraph
            class="c16"
          >
            Idade:
          </paragraph>
          <h2
            class="c13"
          >
            30
          </h2>
        </div>
      </div>
      <div
        class="c14"
      >
        <div
          class="c0 c15"
        >
          <paragraph
            class="c16"
          >
            Telefone:
          </paragraph>
          <h2
            class="c13"
          >
            (272) 790-0888
          </h2>
        </div>
        <div
          class="c0 c15"
        >
          <paragraph
            class="c16"
          >
            Celular:
          </paragraph>
          <h2
            class="c13"
          >
            (489) 330-2385
          </h2>
        </div>
        <div
          class="c0 c15"
        >
          <paragraph
            class="c16"
          >
            Data de registro:
          </paragraph>
          <h2
            class="c13"
          >
            08/07/2007
          </h2>
        </div>
      </div>
      <div
        class="c0 c12"
      >
        <h2
          class="c13"
        >
          Endereço
        </h2>
      </div>
      <div
        class="c14"
      >
        <div
          class="c0 c15"
        >
          <paragraph
            class="c16"
          >
            País:
          </paragraph>
          <h2
            class="c13"
          >
            United States
          </h2>
        </div>
        <div
          class="c0 c15"
        >
          <paragraph
            class="c16"
          >
            Estado:
          </paragraph>
          <h2
            class="c13"
          >
            Michigan
          </h2>
        </div>
        <div
          class="c0 c15"
        >
          <paragraph
            class="c16"
          >
            Cidade:
          </paragraph>
          <h2
            class="c13"
          >
            Billings
          </h2>
        </div>
      </div>
      <div
        class="c14"
      >
        <div
          class="c0 c15"
        >
          <paragraph
            class="c16"
          >
            Rua:
          </paragraph>
          <h2
            class="c13"
          >
            Valwood Pkwy, 8929
          </h2>
        </div>
        <div
          class="c17"
        >
          <a
            href="https://www.google.com/maps/search/?api=1&query=-69.8246,134.8719
"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              color="#fafa"
              fill="currentColor"
              height="28"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgba(255, 170, 255, 0.667);"
              viewBox="0 0 24 24"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.5,6.5 L23,9 L23,22 L16,19 L8,22 L1,19 L1,6 L6,8 M16,19 L16,12 M8,22 L8,12 M12,16.2727273 C12,16.2727273 6,11.5 6,7 C6,3.25 9,1 12,1 C15,1 18,3.25 18,7 C18,11.5 12,16.2727273 12,16.2727273 Z M13,7 C13,6.44766667 12.5523333,6 12,6 C11.4476667,6 11,6.44766667 11,7 C11,7.55233333 11.4476667,8 12,8 C12.5523333,8 13,7.55233333 13,7 Z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
